modular_skyrat/modules/tribal_extended/code/weapons/bow.dm

```diff
@@ -0,0 +1,70 @@
+/obj/item/gun/ballistic/tribalbow
+	name = "Wooden Bow"
+	desc = "Some sort of primitive projectile weapon used to fire arrows."
+	icon = 'modular_skyrat/modules/tribal_extended/icons/projectile.dmi'
+	lefthand_file = 'modular_skyrat/modules/tribal_extended/icons/bows_lefthand.dmi'
+	righthand_file = 'modular_skyrat/modules/tribal_extended/icons/bows_righthand.dmi'
+	icon_state = "bow"
+	w_class = WEIGHT_CLASS_BULKY
+	weapon_weight = WEAPON_HEAVY //need both hands to fire
+	force = 8
+	mag_type = /obj/item/ammo_box/magazine/internal/bow
+	fire_sound = 'modular_skyrat/modules/tribal_extended/sound/sound_weapons_bowfire.ogg'
+	slot_flags = ITEM_SLOT_BACK
+	item_flags = NEEDS_PERMIT
+	casing_ejector = FALSE
+	internal_magazine = TRUE
+	trigger_guard = TRIGGER_GUARD_ALLOW_ALL //so ashwalkers can use it
+
+/obj/item/gun/ballistic/tribalbow/shoot_with_empty_chamber()
+	return
+
+/obj/item/gun/ballistic/tribalbow/chamber_round()
```

Comment:
```suggestion
/obj/item/gun/ballistic/tribalbow/chamber_round(keep_bullet = FALSE, spin_cylinder, replace_new_round)
```
These are your missing parameters.

Reply:
Thank you very much for the assistance, been a rough few days and I'm finally going back to this project.